src/game_api/state_structs.hpp

```diff
@@ -815,6 +824,11 @@ struct LiquidPhysics
             LiquidTileSpawnData stagnant_lava_tile_spawn_data;
         };
     };
+    size_t unknown2;
+    size_t unknown3;
+    LiquidLake* start_lakes;
+    LiquidLake* end_lakes;
+    size_t unknown6;
```

Comment:
That's a vector

also can you add the the next one as well? `uint32 total_liquid_spawned;` it's "total number of spawned liquid entities, all types (not impostor's)"

Reply:
A `game_vector` (using `malloc` and `free`) or a `custom_vector` (using their custom mem-heap)?

Reply:
what's the difference?

it should only matter if you want to add/remove stuff from it, for just from it, it can be simply `std::vector`

Reply:
I updated this to be a std::vector, and also added the rest of the values from the plugin (even though most of them are unknowns).

Reply:
Yeah sure, you can just keep it a `std::vector` but one day when someone decides to use it they might not think about what this really is and break the game in subtle ways. So imho it's worth it to quickly check what this really is when adding it. You just check if it calls `malloc` or not during a `push_back`.

Reply:
I looked a bit, and I have no idea what it's doing to allocate more space. I think it's happening at around 229b7478, but I don't understand it.

Reply:
Updated to use custom_vector based on your recommendation of the alloc at the line I linked.